components/LoginModal.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import { Alert, Modal, Text, TextInput, TouchableHighlight, View } from 'react-native';
+import Styles from './Styles';
+
+export default class LoginModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalVisible: false,
+      loginPressStatus: false,
+      cancelPressStatus: false,
+    };
+  }
+
+  changeVisibility() {
+    this.setState(prevState => ({
+      modalVisible: !prevState.modalVisible,
+    }));
+  }
+
+  render() {
+    return (
+      <View>
+        <Modal
+          id="modal"
+          animationType="fade"
+          presentationStyle="fullScreen"
+          transparent={false}
+          visible={this.state.modalVisible}
+          onRequestClose={() => {
+            Alert.alert('Modal closed');
+          }}
+        >
+          <View>
+            <TextInput
```

Comment:
Consider adding textContentType as username to the input as well. Will let users autofill from eg icloud keychain or other password manager.

Reply:
Sorted